packages/mobile/src/paymentRequest/IncomingPaymentRequestListItem.tsx

```diff
@@ -22,25 +24,57 @@ interface OwnProps {
   comment: string
   id: string
   declinePaymentRequest: typeof declinePaymentRequest
+  fetchPhoneAddressesAndCheckIfRecipientValidationRequired: typeof fetchPhoneAddressesAndCheckIfRecipientValidationRequired
+  manualAddressValidationRequired: boolean
+  fullValidationRequired: boolean
 }
 
 type Props = OwnProps & WithTranslation
 
 const AVATAR_SIZE = 40
 
 export class IncomingPaymentRequestListItem extends React.Component<Props> {
+  componentDidMount() {
+    // need to check latest mapping to prevent user from accepting fradulent requests
+    this.fetchLatestPhoneAddressesAndRecipientVerificationStatus()
+  }
+
+  fetchLatestPhoneAddressesAndRecipientVerificationStatus = () => {
+    const { requester: recipient } = this.props
+
+    // Skip phone number fetch for QR codes or Addresses
+    if (recipient.kind !== RecipientKind.QrCode && recipient.kind !== RecipientKind.Address) {
```

Comment:
I would find this easier to read if it was checking whether the RecipientKind matched one that required a phone number rather than skipping it for ones that do not